Client/Frontend/Settings/SettingsTableViewController.swift

```diff
@@ -184,6 +184,33 @@ private class DisconnectSetting: WithAccountSetting {
     }
 }
 
+private class SyncNowSetting: WithAccountSetting {
+    override var accessoryType: UITableViewCellAccessoryType { return .None }
+
+    override var style: UITableViewCellStyle { return .Value1 }
+
+    override var title: NSAttributedString? {
+        return NSAttributedString(string: NSLocalizedString("Sync Now", comment: "Sync Firefox Account"), attributes: [NSForegroundColorAttributeName: UIConstants.HighlightBlue])
+    }
+
+    private override func onConfigureCell(cell: UITableViewCell) {
+        cell.detailTextLabel?.attributedText = status
+        cell.detailTextLabel?.font = cell.detailTextLabel?.font.fontWithSize(11)
+        cell.textLabel?.attributedText = title
+        cell.accessoryType = accessoryType
+        cell.accessoryView = nil
+    }
+    
+    override func onClick(navigationController: UINavigationController?) {
+        profile.syncManager.syncEverything().uponQueue(dispatch_get_main_queue()) { result in
```

Comment:
Perhaps instead of a spinner we should change the label to "Syncing…" and disable the button. Otherwise you can hammer the button.